### PR TITLE
p2p: don't connect to more than one IP per class B if we can

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/p2p/net_node.inl [hunk 1]
```diff
@@ -1226,19 +1226,53 @@ namespace nodetool
...
+      std::set<uint32_t> classB;
+      if (&zone == &m_network_zones.at(epee::net_utils::zone::public_)) // at returns reference, not copy
+      {
+        zone.m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt)
+        {
+          if (cntxt.m_remote_address.get_type_id() == epee::net_utils::ipv4_network_address::get_type_id())
+          {
+
+            const epee::net_utils::network_address na = cntxt.m_remote_address;
+            const uint32_t actual_ip = na.as<const epee::net_utils::ipv4_network_address>().ip();
```
Comment (hyc) [inline src/p2p/net_node.inl]: Does epee always return IP addresses in little-endian byte order, or is this machine-specific?

Comment (moneromooo-monero) [reply src/p2p/net_node.inl]: It appears so. See https://github.com/monero-project/monero/pull/5544/commits/04a103be2d5174b3febadd296215a9cf2b3a10dc, and I'm noticing that the comment there is wrong, they're kept little endian, not network byte order. It was confusing to get it right...

Comment (hyc) [reply src/p2p/net_node.inl]: That commit doesn't give me much confidence - it looks like addresses are kept in machine-specific byte order, and the rest of the code then treats it as little-endian, meaning that there are more cases to fix on big-endian machines.

Comment (moneromooo-monero) [reply src/p2p/net_node.inl]: I would totally accept a patch which straightens all this out :)

Still, monerod syncs fine with the patches in the PR the link above is from, so it's progress.

**Pull request merged**